src/AlignLoads.cpp

```diff
@@ -92,8 +92,7 @@ class AlignLoads : public IRMutator {
             int shift = known_alignment && aligned_offset < stride ? aligned_offset : 0;
 
             // Load a dense vector covering all of the addresses in the load.
-            internal_assert(stride >= 0);
-            Expr dense_base = simplify(ramp->base - aligned_offset);
+            Expr dense_base = simplify(ramp->base - shift);
```

Comment:
Thanks for fixing this. I had noticed the inconsistency in the use of aligned_offset here and shift below in the making of the index. I thought it was suspicious, but some how convinced myself that it was ok. I guess I was wrong :)